pkg/utils/validation/admissionplugins/admissionplugins.go

```diff
@@ -85,6 +85,12 @@ var (
 
 	admissionPluginsSupportingExternalKubeconfig = sets.New("ValidatingAdmissionWebhook", "MutatingAdmissionWebhook", "ImagePolicyWebhook")
 
+	// These plugins can be specified in the Shoot spec even if the version doesn't support it. This is required to facilitate migration of these plugins in some cases.
+	// For example, the "PodSecurityPolicy" plugin should be disabled in the Shoot spec for an upgrade from Kubernetes v1.24 to v1.25, but in v1.25 this plugin is not
+	// supported. gardener-apiserver will/should take care to clean this plugin from the spec, because otherwise this will be passed to the kube-apiserver as a flag and
+	// kube-apiserver will panic. See https://github.com/gardener/gardener/pull/8212 for more details.
+	pluginsInMigration = sets.New("PodSecurityPolicy")
```

Comment:
Should we make this a `map[string]*semver.Constraint`, and export and reuse it in https://github.com/gardener/gardener/blob/83ff54fca09181e3b974315a4c8200526259c828/pkg/registry/core/shoot/strategy.go#L196 and https://github.com/gardener/gardener/blob/83ff54fca09181e3b974315a4c8200526259c828/pkg/registry/core/shoot/strategy.go#L184?

Reply:
Nice idea, I adapted it in [f2f7433](https://github.com/gardener/gardener/pull/8212/commits/f2f743358856fff38d6b8377400561e41247a5c1)

Reply:
Ideally the constraint should have been `=1.25`, but now maybe we should not change it and break existing clusters.